src/main/java/com/tabnine/UninstallListener.kt

```diff
@@ -0,0 +1,62 @@
+package com.tabnine
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor
+import com.intellij.ide.plugins.PluginStateListener
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.util.text.SemVer
+import com.tabnine.binary.BinaryRequestFacade
+import com.tabnine.binary.requests.uninstall.UninstallRequest
+import com.tabnine.general.StaticConfig
+import com.tabnine.general.readTempTabninePluginZip
+import com.tabnine.lifecycle.UninstallReporter
+import java.nio.file.Paths
+
+private const val TABNINE_JAR_NAME = "TabNine-"
+private const val JAR_SUFFIX = ".jar"
+private val TABNINE_JAR_REGEX = """^$TABNINE_JAR_NAME\d+\.\d+\.\d+.jar""".toRegex()
```

Comment:
sometimes we release with pre-release - 0.8.0-alpha20220512.....

Probably account for that, ormaybe just check it starts with TabNine- or something

Reply:
i'd rather account for the pre-releases with `...*.jar`, so that i know for sure i can extract the semver from the `d+.d+.d+` section